File: scala-play/app/services/CheckService.scala
package services.check

import javax.inject._
import scala.concurrent._
import models._

@Singleton
class CheckService @Inject()()(implicit ec: ExecutionContext) extends ICheckService {
  override def checkEmpty(): Future[CheckEmptyResponse] = Future {
    CheckEmptyResponse.Ok()
  }

  override def checkForbidden(): Future[CheckForbiddenResponse] = Future {
    CheckForbiddenResponse.Forbidden()
  }

  override def sameOperationName(): Future[SameOperationNameResponse] = Future {
    SameOperationNameResponse.Ok()
  }
}

File: ts-express-superstruct/src/check_service.ts
import * as service from './spec/check'
import * as models from './spec/models'

export let checkService = (): service.CheckService => {
    let checkEmpty = async (): Promise<void> => {}

    let checkEmptyResponse = async (params: service.CheckEmptyResponseParams): Promise<void> => {}
    
    let checkForbidden = async (): Promise<service.CheckForbiddenResponse> => {
        return {status: 'forbidden'}
    }

    let sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        return {status: 'ok'}
    }

    return {checkEmpty, checkEmptyResponse, checkForbidden, sameOperationName}
}